src/main/scala/com/atamaniv/task_3/task3.scala

```diff
@@ -0,0 +1,186 @@
+package com.atamaniv.task_3
+
+import cats.{Id, Monad}
+
+import scala.concurrent.Future
+
+/**
+  * Repository and Service implementation using tagless final pattern.
+  * The idea is to make it easier to test our database layer, using Scala’s higher kinded types to abstract
+  * the Future type constructor away from our traits under test.
+  * Intro to tagless final: https://www.basementcrowd.com/2019/01/17/an-introduction-to-tagless-final-in-scala/.
+  * The similar task example https://github.com/LvivScalaClub/cats-playground/blob/master/src/main/scala/BookRepository.scala
+  */
+
+case class User(id: Long, username: String)
+
+case class IotDevice(id: Long, userId: Long, sn: String)
+
+// NOTE: This import bring into the scope implicits that allow you to call .map and .flatMap on the type F[_]
+// and also bring you typeclasses that know how to flatmap (Monad) and map (Functor) over your higher-kinded type.
+import cats.implicits._
+
+trait UserRepository[F[_]] {
```

Comment:
It would be nice to put that in different file and package. Task description have it in one just for requirements purpose.